sklearn/ensemble/tests/test_iforest.py

```diff
@@ -129,6 +130,12 @@ def test_iforest_error():
     with pytest.raises(NotImplementedError, match=msg):
         IsolationForest(behaviour='old').fit(X)
 
+    # test feature_weight length matches n_features:
+    assert_raises(ValueError, IsolationForest(feature_weight=[1]).fit, X)
+    # test feature_weight adding up to 1:
```

Comment:
This is not a constraint we place on sample_weight... Are we sure we should require it here??

Reply:
Got it. I change it to normalize the feature weight in `BaseBagging` because `np.random.choice` requires probability.